### DISCUSSION TITLE
Multicores processing

### DISCUSSION THREAD

**Discussion by AKHCE** [MilesCranmer/PySR#251] [Q&A] [answered]
Hi,
Although I have many cores available, but the PySR is not using more than a few cores (2-4 cores). Is there a way to make PySR to use all cores available? I'd like to add that my search is really complex (populations, iterations, generations are big numbers), for example, after running PySR for 4 days (4 cores used max), I succeed to get close to the desired results. I am trying to find a way PySR uses more cores 
I don't pass proc to PySRRegressor.

Any advice is much appreciated, 
Ahmad

**Comment by MilesCranmer** [comment, marked as answer]
The `procs` argument controls how many cores it uses. So for using 4 cores you would set `procs=4`.

If it still doesn't use all the cores, then it could be that the main thread is doing too much work. In that case, try increasing `ncyclesperiteration` until all cores are being used. Also, having `populations >= 2 * procs` is a good idea, so that all threads can be occupied.

You could also try using multiprocessing instead of multithreading. To do that, use `multithreading=False, procs=4`.

**Reply by MilesCranmer** [reply]
For details, see this page: https://astroautomata.com/PySR/api/#performance-and-parallelization

**Reply by AKHCE** [reply]
Would you please explain what ncyclesperiteration does in non-very-technical words?
Many thanks

**Reply by MilesCranmer** [reply]
Larger ncyclesperiteration = threads spend more time working before talking to the leader thread
